astarte/device/mapping.py

```diff
@@ -139,16 +139,24 @@ def __init__(self, mapping_definition: dict, interface_type: str):
         interface_type:
             Type of the parent Interface, used to determine the default reliability
         """
-        self.endpoint: str = mapping_definition["endpoint"]
-        self.type: str = mapping_definition["type"]
-        self.__actual_type = type_strings.get(mapping_definition["type"])
+        self.endpoint: str = mapping_definition.get("endpoint")
+        self.type: str = mapping_definition.get("type")
+        self.__actual_type = type_strings.get(mapping_definition.get("type"))
         self.explicit_timestamp = mapping_definition.get("explicit_timestamp", False)
-        self.reliability = 2
-        if interface_type == "datastream":
-            if "reliability" in mapping_definition:
-                self.reliability = QOS_MAP[mapping_definition["reliability"]]
-            else:
-                self.reliability = 0
+        default_reliability = "unreliable" if interface_type == "datastream" else "unique"
+        self.reliability = QOS_MAP.get(mapping_definition.get("reliability", default_reliability))
+        self.allow_unset = mapping_definition.get("allow_unset", False)
+
+        if not (isinstance(self.endpoint, str) and self.__actual_type):
+            raise InterfaceFileDecodeError("Error parsing the mapping.")
+        if (interface_type != "datastream") and any(
+            k in mapping_definition for k in ("explicit_timestamp", "reliability")
+        ):
+            raise InterfaceFileDecodeError(
+                "Fields 'reliability' and 'explicit_timestamp' have no meaning for properties."
+            )
+        if ("allow_unset" in mapping_definition) and (interface_type == "datastream"):
+            raise InterfaceFileDecodeError("Field 'allow_unset' have no meaning for datastreams.")
```

Comment:
Those checks could be inverted, like `if "allow_unset" and not "property"`, but it wouldn't change much for me